vue2-vuetify/src/dynaform/DateTimeControlRenderer.vue

```diff
@@ -43,64 +67,172 @@ import {
+    VBtn,
     CustomControlWrapper,
   },
   props: {
     ...rendererProps<ControlElement>(),
   },
   setup(props: RendererProps<ControlElement>) {
+    let date = ref(dateNow());
+    let dateFormatted = ref(formatDate(dateNow()));
+    let time = ref(timeNow());
+    let timeFormatted = ref(formatTime(timeNow()));
+    let dateTime = ref(dateTimeNow().replace('T', ' '));
+    let dateTimeFormatted = ref(
+      formatDateTime.apply(this, [dateTimeNow().replace('T', ' ')])
+    );
+
+    function blurChangeFormat(this: any) {
+      if (this.inputFormat === 'date') {
+        this.date = parseDate(
+          'value' in dateFormatted ? dateFormatted.value : dateFormatted
+        );
+      } else if (this.inputFormat === 'time') {
+        this.time = formatTime(
+          'value' in timeFormatted ? timeFormatted.value : timeFormatted
+        );
+      } else if (this.inputFormat === 'date-time') {
+        this.dateTime = parseDateTime(
+          'value' in dateFormatted ? dateFormatted.value : dateFormatted,
+          'value' in timeFormatted ? timeFormatted.value : timeFormatted
+        );
+      }
+    }
+
     return {
       ...useVuetifyControl(useJsonFormsControl(props)),
+      menu: false,
+      date,
+      dateFormatted,
+      time,
+      timeFormatted,
+      dateTime,
+      dateTimeFormatted,
+      formatDate,
+      parseDate,
+      formatTime,
+      formatDateTime,
+      parseDateTime,
+      blurChangeFormat,
     };
   },
+  watch: {
+    date() {
+      this.dateFormatted = this.formatDate(this.date);
+      if (this.inputFormat === 'date-time') {
+        this.dateTimeFormatted = this.formatDateTime(
+          this.date + ' ' + this.time
+        );
+      }
+    },
+    time() {
+      this.timeFormatted = this.formatTime(this.time);
+      if (this.inputFormat === 'date-time') {
+        this.dateTimeFormatted = this.formatDateTime(
+          this.date + ' ' + this.time
+        );
+      }
+    },
+    dateTime() {
+      this.dateTimeFormatted = this.formatDateTime(this.dateTime);
+    },
+  },
   computed: {
     hint(): string {
       return this.control.uischema.options?.hint ?? '';
     },
     placeholder(): string {
       return this.control.uischema.options?.placeholder ?? '';
     },
-    dataTime: {
-      get(): string | null | undefined {
-        const datetimeLocalFormat = 'YYYY-MM-DDTHH:mm:ss.SSS';
-        const saveFormat = this.appliedOptions.dateTimeSaveFormat ?? undefined;
-        const value = this.control.data as string | undefined | null;
+    inputFormat(): string | undefined {
+      return this.control.schema?.format ?? '';
+    },
+    valueChangeFormat(): string | null {
+      if (this.inputFormat === 'date') {
+        return this.dateFormatted;
+      } else if (this.inputFormat === 'time') {
+        return this.timeFormatted;
+      } else if (this.inputFormat === 'date-time') {
+        return this.dateTimeFormatted;
+      } else {
+        return null;
+      }
```

Comment:
Remove the statement else, is not necessary

Reply:
Removing this else statement renders the following error:
`Error: /home/martin/Documents/dunamis-jform/vue2-vuetify/src/dynaform/DateTimeControlRenderer.vue?rollup-plugin-vue=script.ts(180,26): semantic error TS2366: Function lacks ending return statement and return type does not include 'undefined'`
Can you suggest a fix for this please?